src/NATS.Client.KeyValueStore/Internal/NatsKVWatcher.cs

```diff
@@ -368,6 +367,18 @@ private async ValueTask<INatsJSConsumer> CreatePushConsumer(string origin)
             ReplayPolicy = ConsumerConfigReplayPolicy.Instant,
         };
 
+        // nat2.9.x
+        if (_filters.Length == 1)
```

Comment:
if this is not the case and server version is <2.10 should we error?

Reply:
Are there any NATS policies on how many versions you want to support? I ask, because if 2.11 is around the corner, maybe it's a moot point. I think for convenience it'd be nice to at least log an error, but since there is a chance that the functionality will differ from expectations, I think an exception makes sense.

If one of y'all end up adding the additional exception handling, please feel free to fix my two typos (`// nat2.9.x`). My OCD is triggered.

Reply:
Technically the NATS Server hosting JS must be 2.10, not necessarily the NATS Server it is connected to. I think if multiple filter subjects are passed and the server doesn't support it, it will see that as no filter subjects and throw an error, which could be confusing though.

Reply:
> Are there any NATS policies on how many versions you want to support? I ask, because if 2.11 is around the corner, maybe it's a moot point.

I think we should stick to >=2.9 for the time being but you're right we will have to think about it as versions go up.

> Technically the NATS Server hosting JS must be 2.10, not necessarily the NATS Server it is connected to

good point. so there is not use trying to check server version.